draft-irtf-cfrg-hpke.md

```diff
@@ -830,16 +890,56 @@ security models and assumptions, and do not consider attackers capable of quantu
 computation. A full proof of post-quantum security would need to take this
 difference into account, in addition to simply using a post-quantum KEM.
 
-## Domain Separation
-
-\[\[RFC editor: please change "RFCXXXX" to the correct number before
-publication.]]
-
-The KeySchedule procedure includes the domain separation string "RFCXXXX" in
-each Expand invocation. This ensures any secrets derived in HPKE are independent
-from those used in other protocols, even when derived from the same IKM (secret).
-Derivation of the KeySchedule 'secret' does not include domain separation as it
-is an intermediate value not exposed by the protocol.
+## Pre-Shared Key Recommendations {#security-psk}
+
+In mode PSK and mode AuthPSK, the PSK SHOULD be of length Nh bytes or
+longer, and SHOULD have Nh bytes of entropy or more. Using a PSK shorter
+than Nh bytes is permitted. A PSK that is longer than Nh bytes or that
+has more than Nh bytes of entropy, respectively, does not increase the
+security level of HPKE, because the extraction step involving the PSK
+only outputs Nh bytes.
+
+HPKE is specified to use HKDF as key derivation function. HKDF is not
+designed to slow down dictionary attacks, see {{?RFC5869}}. Thus,
+HPKE's PSK mechanism is not suitable for use with a low-entropy
+password as the PSK: in scenarios in which the adversary knows the
+KEM shared secret zz and has access to an oracle for decryption success
+of an HPKE ciphertext, it can perform a dictionary attack on the PSK.
+If an application does not use Seal to create HPKE ciphertexts but only
```

Comment:
How does Seal/Open prevent this?  If you behave differently on decryption failure, then it seems like you still create an oracle.

Reply:
New proposal:

Thus, HPKE's PSK mechanism is not suitable for use with a low-entropy
password as the PSK: in scenarios in which the adversary knows the
KEM shared secret zz and has access to an oracle that allows to distinguish
between a good and a wrong key, it can perform a dictionary attack on the PSK.
This oracle can be the decryption operation on a captured HPKE ciphertext or
any other recipient behavior which is observably different when using a wrong key.

Reply:
Works for me, thanks.